core/indexer/elasticsearchDatabase_test.go

```diff
@@ -0,0 +1,317 @@
+package indexer
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"math/big"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/core/mock"
+	"github.com/ElrondNetwork/elrond-go/data"
+	dataBlock "github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/logger"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+	"github.com/stretchr/testify/require"
+)
+
+type databaseWriterMock struct {
+	doRequestCalled     func(req esapi.IndexRequest) error
+	doBulkRequestCalled func(buff *bytes.Buffer, index string) error
+}
+
+func (dwm *databaseWriterMock) doRequest(req esapi.IndexRequest) error {
+	if dwm.doRequestCalled != nil {
+		return dwm.doRequestCalled(req)
+	}
+	return nil
+}
+
+func (dwm *databaseWriterMock) doBulkRequest(buff *bytes.Buffer, index string) error {
+	if dwm.doBulkRequestCalled != nil {
+		return dwm.doBulkRequestCalled(buff, index)
+	}
+	return nil
+}
+
+func (dwm *databaseWriterMock) checkAndCreateIndex(_ string, _ io.Reader) error {
+	return nil
+}
+
+func newTestElasticSearchDatabase(elasticsearchWriter databaseWriterHandler, arguments elasticSearchDatabaseArgs) *elasticSearchDatabase {
+	return &elasticSearchDatabase{
+		client:      elasticsearchWriter,
+		marshalizer: arguments.marshalizer,
+		hasher:      arguments.hasher,
+	}
+}
+
+func createMockElasticsearchDatabaseArgs() elasticSearchDatabaseArgs {
+	return elasticSearchDatabaseArgs{
+		url:         "url",
+		userName:    "username",
+		password:    "password",
+		hasher:      &mock.HasherMock{},
+		marshalizer: &mock.MarshalizerMock{},
+	}
+}
+
+func newTestTxPool() map[string]data.TransactionHandler {
+	txPool := make(map[string]data.TransactionHandler)
+
+	txPool["tx1"] = &transaction.Transaction{
+		Nonce:     uint64(1),
+		Value:     big.NewInt(1),
+		RcvAddr:   []byte("receiver_address1"),
+		SndAddr:   []byte("sender_address1"),
+		GasPrice:  uint64(10000),
+		GasLimit:  uint64(1000),
+		Data:      []byte("tx_data1"),
+		Signature: []byte("signature1"),
+	}
+
+	txPool["tx2"] = &transaction.Transaction{
+		Nonce:     uint64(2),
+		Value:     big.NewInt(2),
+		RcvAddr:   []byte("receiver_address2"),
+		SndAddr:   []byte("sender_address2"),
+		GasPrice:  uint64(10000),
+		GasLimit:  uint64(1000),
+		Data:      []byte("tx_data2"),
+		Signature: []byte("signature2"),
+	}
+
+	txPool["tx3"] = &transaction.Transaction{
+		Nonce:     uint64(3),
+		Value:     big.NewInt(3),
+		RcvAddr:   []byte("receiver_address3"),
+		SndAddr:   []byte("sender_address3"),
+		GasPrice:  uint64(10000),
+		GasLimit:  uint64(1000),
+		Data:      []byte("tx_data3"),
+		Signature: []byte("signature3"),
+	}
+
+	return txPool
+}
+
+func newTestBlockBody() dataBlock.Body {
+	return dataBlock.Body{
+		{TxHashes: [][]byte{[]byte("tx1"), []byte("tx2")}, ReceiverShardID: 2, SenderShardID: 2},
```

Comment:
consider having "tx1", "tx2" and such as unexproted consts in this file named "tx1Hash", "tx2Hash"....